django_serializer/swagger/utils.py

```diff
@@ -0,0 +1,87 @@
+from django import forms
+from django.forms import ModelForm, modelform_factory
+from marshmallow import Schema, fields
+
+from django_serializer.v2.exceptions import HttpError
+
+FORM_FIELD_MAPPING = {
+    forms.IntegerField: fields.Int,
+    forms.BooleanField: fields.Bool,
+    forms.CharField: fields.Str,
+    forms.DateField: fields.Date,
+    forms.DateTimeField: fields.DateTime,
+    forms.TimeField: fields.Time,
+    forms.DecimalField: fields.Decimal,
+    forms.EmailField: fields.Str,
+    forms.FloatField: fields.Float,
+    # forms.ImageField: ImageField,
+    forms.URLField: fields.Str
+}
+
+
+def form2schema(form: forms.Form) -> Schema:
+    if form is None:
+        return None
+
+    if issubclass(form, ModelForm):
+        # ModelForm по умолчанию не инициализирована, поэтому надо применить
+        # modelform_factory, чтобы получить поля
+        form = modelform_factory(form.model, fields=form.fields)
+        form_fields = form.base_fields
+    else:
+        form_fields = form.declared_fields
+
+    schema = schema_factory(str(id(form)), form_fields.keys())
+
+    schema_fields = {}
+    for name, form in form_fields.items():
+        schema_fields.update(
+            {name: FORM_FIELD_MAPPING[type(form)](required=form.required)})
+    schema = schema(**schema_fields)
+    return schema
+
+
+def schema_factory(name, arg_names, base_class=Schema):
```

Comment:
ma already have mechanism for creating schema from dict https://marshmallow.readthedocs.io/en/stable/quickstart.html#creating-schemas-from-dictionaries